odoo/models.py

```diff
@@ -3114,6 +3114,7 @@ def unlink(self):
             Data = self.env['ir.model.data'].sudo().with_context({})
             Defaults = self.env['ir.default'].sudo()
             Attachment = self.env['ir.attachment']
+            Translation = self.env['ir.translation']
```

Comment:
You have already checked that you can delete ``self``, so there is no point checking again for translations. Doing the ``unlink()`` as the superuser will save two SQL queries. Definitively worth it.
```suggestion
            Translation = self.env['ir.translation'].sudo()
```
We could alternatively make a private method that discards the translations with a single SQL query...